### ISSUE TITLE
Add attribution to full installer downloads

### ISSUE THREAD

**Issue by hoosteeno** [mozilla/bedrock#8910]
## Description

Work is underway to add attribution telemetry to full installers. Bedrock needs to be modified to send attribution parameters on full installer downloads. This work requires some conversation about any rollout constraints (does Bedrock need to come first? last? do we need staging for QA?), and may (unlikely) require additional parameters. Meanwhile, it can get started, but *do not merge*.

---

#### :yellow_heart: Success Criteria :yellow_heart:
- [ ] This card or another describes the rollout plan
- [ ] All parameters required are specified in this card or another
- [ ] Full installer downloads include attribution parameters that work in telemetry attribution pipeline

**Comment by hoosteeno** [comment]
Here's a search that can find all the full installer download locations, in case they need to be changed:

https://github.com/mozilla/bedrock/search?q=force_full_installer%3DTrue&unscoped_q=force_full_installer%3DTrue

**Comment by alexgibson** [comment]
@hoosteeno is there an associated bug in bugzilla for more context here?

**Comment by tublitzed** [comment]
the issue to include within the installer is [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1316136).

overarching plan for context is [here](https://docs.google.com/document/d/13lAPjcnBJvazy1NwwSe4oLCTuN114UlhM0P9lV1h7CY/edit?ts=5e85b34d#)

**Comment by alexgibson** [comment]
Ok I've done some preliminary investigation here and can confirm the following:

- We're already doing attribution on full installer links. What I mean is, as far as bedrock is concerned we don't make any distinction if a download button is either full installer or stub installer. If you visit /firefox/installer-help/ in production right now, on a Windows machine, attribution is already working on those full installer links.

That said, I have found some edgecases we're not yet covering:

- We're not doing any kind of attribution on /firefox/all/, either stub or full installer. This is because that page uses some custom code for the download button.
- In the modal on /firefox/new/, we're doing attribution for windows 64bit and 32bit builds, but we're not covering the MSI installers. I think this is likely due to MSI only recently being added to the website.

If I fix the two scenarios above, then I think we're all done as far as bedrock goes.

/cc @hoosteeno

**Comment by alexgibson** [comment]
Another area where we're missing attribution (and as far as I can tell custom GA events entirely) is on the `/firefox/enterprise/` page.

**Comment by alexgibson** [comment]
Demo URLs (test on Windows):

- https://www-demo3.allizom.org/en-US/firefox/all/ (attribution is applied on download button click only)
- https://www-demo3.allizom.org/en-US/firefox/new/ (attribution is applied on page load to direct download links in modal)
- https://www-demo3.allizom.org/en-US/firefox/enterprise/ (attribution is applied on page load to direct download links in modal)

**Comment by alexgibson** [comment]
@hoosteeno given the above, we should discuss the possibility of rolling out these fixes independently of any work happening in bouncer / stub installer to support this.

**Comment by hoosteeno** [comment]
@tublitzed since we have been sending attribution params with many full installer downloads already, it seems we don't need to wait on any downstream work before rolling this out. Do you agree?

**Comment by hoosteeno** [comment]
I've shared the demo links with the product team. Still holding for any info about https://github.com/mozilla/bedrock/issues/8910#issuecomment-630298230.

**Comment by alexgibson** [comment]
Adding a blocker to this until we hear back.

**Comment by tublitzed** [comment]
Ah, missed the notification. Sorry for the delay here. @hoosteeno and @alexgibson - Agreed, I don't think we need to wait to roll this out if we've already been doing it elsewhere without issue. 👍 from me here.

**Comment by hoosteeno** [comment]
This issue is now blocked by #8963, but both issues are unblocked for coding and merging together.

**Comment by alexgibson** [comment]
@hoosteeno can you explain why this issue is blocked by #8963 (technically)? I'm not quite following why one needs to depend on the other. Thanks

**Comment by hoosteeno** [comment]
I don't think it is technically blocked, but I do want to understand the plan for attribution system changes before we call this done, and #8963 is on the critical path for that.

**Comment by alexgibson** [comment]
Ok, seeing as https://github.com/mozilla/bedrock/issues/8963 is a wontfix, I'm going to unblock this and move it along to code review.

**Comment by hoosteeno** [comment]
Firefox team has QA'd this, so it's ready for Bedrock dev review and merging.

**Comment by jchaupitre** [comment]
Hi Justin, not clear to me who is the firefox team who QAed this, are you saying that this PI here has been fully tested and now complete https://jira.mozilla.com/browse/PI-641? if so, I will ask them to indicate so and sign-off. thanks!

**Comment by alexgibson** [comment]
@jchaupitre I helped guide tzsoldos through testing the bedrock parts here last week, although I can't speak to the other dependencies. I'm not sure if that needs to block us moving forward with the change here though, seeing as the functionality is essentially already in producton (See https://github.com/mozilla/bedrock/issues/8910#issuecomment-629241503)

**Comment by jchaupitre** [comment]
Got it, thanks. I just followed up with QA on the PI request.
